### PR TITLE
Renamed GOPath functions and the file

### PR DESCRIPTION
**Pull request by oleg68**
As discusses in #1991, instead of renaming one function, this PR renames all the global functions in the GOPath file according to the current GO naming convention.

It also renames the GOPath files themself.

It is just renaming. No GO behavior should be changed.

### CODE REVIEW

File: src/core/go_path.cpp [hunk 1]
```diff
@@ -1,44 +1,44 @@
...
 
-wxString GOGetPath(const wxString &path) {
-  wxFileName name(GONormalizePath(path));
+wxString go_get_path(const wxString &path) {
+  wxFileName name(go_normalize_path(path));
   return name.GetPath();
 }
 
-void GOSyncDirectory(const wxString &path) {
+void go_synch_directory(const wxString &path) {
```
Comment (larspalo) [inline src/core/go_path.cpp:33]:
```suggestion
void go_sync_directory(const wxString &path) {
```

File: src/core/go_path.cpp [hunk 2]
```diff
@@ -50,6 +50,6 @@ bool GORenameFile(const wxString &from, const wxString &to) {
       return false;
     }
   }
-  GOSyncDirectory(name.GetPath());
+  go_synch_directory(name.GetPath());
```
Comment (larspalo) [inline src/core/go_path.cpp:53]:
```suggestion
  go_sync_directory(name.GetPath());
```

File: src/core/go_path.h [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+#ifndef GOPATH_H
+#define GOPATH_H
+
+#include <wx/string.h>
+
+void go_create_directory(const wxString &path);
+wxString go_normalize_path(const wxString &path);
+wxString go_get_path(const wxString &path);
+
+void go_synch_directory(const wxString &path);
```
Comment (larspalo) [inline src/core/go_path.h:17]:
```suggestion
void go_sync_directory(const wxString &path);
```

File: src/grandorgue/GOAudioRecorder.cpp [hunk 4]
```diff
@@ -126,7 +126,7 @@ void GOAudioRecorder::StopRecording() {
   m_recorder->Close();
   if (!m_DoRename) {
     wxFileName name = m_Filename;
-    GOSyncDirectory(name.GetPath());
+    go_synch_directory(name.GetPath());
```
Comment (larspalo) [inline src/grandorgue/GOAudioRecorder.cpp:129]:
```suggestion
    go_sync_directory(name.GetPath());
```

File: src/grandorgue/midi/GOMidiRecorder.cpp [hunk 5]
```diff
@@ -318,7 +318,7 @@ void GOMidiRecorder::StopRecording() {
   m_file.Close();
   if (!m_DoRename) {
     wxFileName name = m_Filename;
-    GOSyncDirectory(name.GetPath());
+    go_synch_directory(name.GetPath());
```
Comment (larspalo) [inline src/grandorgue/midi/GOMidiRecorder.cpp:321]:
```suggestion
    go_sync_directory(name.GetPath());
```

File: src/grandorgue/GOFrame.cpp [hunk 6]
```diff
@@ -1377,11 +1377,11 @@ void GOFrame::OnRenameFile(wxRenameFileEvent &event) {
     wxFD_SAVE | wxFD_OVERWRITE_PROMPT);
   if (dlg.ShowModal() == wxID_OK) {
     if (filepath.GetFullPath() != dlg.GetPath())
-      GORenameFile(filepath.GetFullPath(), dlg.GetPath());
+      go_rename_file(filepath.GetFullPath(), dlg.GetPath());
   } else
     wxRemoveFile(filepath.GetFullPath());
 
-  GOSyncDirectory(filepath.GetPath());
+  go_synch_directory(filepath.GetPath());
```
Comment (larspalo) [inline src/grandorgue/GOFrame.cpp:1384]:
```suggestion
  go_sync_directory(filepath.GetPath());
```

**Review by larspalo — CHANGES_REQUESTED** [review]
I still think that "sync" is a better option as a word than "synch" - which rather should have been "synchronize" but that's perhaps too long.

File: src/core/go_path.cpp (hunk 1 above)
Comment (oleg68) [reply src/core/go_path.cpp]: Done

File: src/core/go_path.h (hunk 3 above)
Comment (oleg68) [reply src/core/go_path.h]: Done

File: src/grandorgue/midi/GOMidiRecorder.cpp (hunk 5 above)
Comment (oleg68) [reply src/grandorgue/midi/GOMidiRecorder.cpp]: Done

File: src/grandorgue/GOFrame.cpp (hunk 6 above)
Comment (oleg68) [reply src/grandorgue/GOFrame.cpp]: Done

File: src/grandorgue/GOAudioRecorder.cpp (hunk 4 above)
Comment (oleg68) [reply src/grandorgue/GOAudioRecorder.cpp]: Done

File: src/core/go_path.cpp (hunk 2 above)
Comment (oleg68) [reply src/core/go_path.cpp]: Done

**Comment by oleg68** [comment]
@rousseldenis could you approve this PR?

**Review by rousseldenis — APPROVED** [review]

**Pull request merged**